### ISSUE TITLE
[Bug Report / Proposal] env.seed doesn't work and no method to get a new RNG

### ISSUE THREAD

**Issue by pseudo-rnd-thoughts** [openai/gym#2856]
Edit: `env.seed` is working, I just discovered a bug with wrappers and `_np_random`, see #2857 for the fix. 

`env.seed(None)` doesn't work
```
import gym

env = gym.make('CartPole-v1', disable_env_checker=True)
print(f'pre-env.seed {env._np_random=}')  # pre-env.seed env._np_random=None
env.seed(None)
print(f'post-env.seed {env._np_random=}')  # post-env.seed env._np_random=None

print(f'pre-env.np_random {env._np_random=}')  # pre-env.np_random env._np_random=None
env.np_random
print(f'post-env.np_random {env._np_random=}')  # post-env.np_random env._np_random=RandomNumberGenerator(PCG64) at 0x7FB8F8067D30
```
The problem is that `env.seed` doesn't actually seem the set the seed of the environment even if this is a value not `None`
The reason for this is unclear, I believe it could be because of it be a class attribute not an object attribute but this is some of the weirdness of python
The only way to get a seed is to use `np_random` which is a property function that will assign `_np_random` the true RNG a new generator. 
Maybe Im missing something but this feels like a bug. 
FYI, this is not an issue with `Space` as `_np_random` is an object attribute

The second issue, which is the one I was originally making this issue on is that there is no way to generate a new RNG with the new API, i.e. without `env.seed()`
With the new api for `reset` then when `seed=None`, this is just ignored with the "old" `np_random` being used
This means that without calling `seed(None)` there is currently no explained method of getting a "new" `np_random`

This is an issue when you want to deepcopy an environment that is often done for vectorisation.
Previously this was possible as `gym.make` did not call `reset` or `step` that initialises the `np_random`
Meaning that on the first vectorised call of `reset` then the each environment will create their own unique `np_random`

Would we want to add a way of getting a "new" `np_random` through setting the `seed=-1`, in valid seed?

Edit: It gets worse, `env.seed` doesn't work at all 
```
import gym

env = gym.make('CartPole-v1', disable_env_checker=True)
print(f'pre-env.np_random {env._np_random=}')
# pre-env.np_random env._np_random=None
env.np_random
print(f'post-env.np_random {env._np_random=}')
# post-env.np_random env._np_random=RandomNumberGenerator(PCG64) at 0x7F0BD7557D30

print(f'pre-env.seed np_random state: {env._np_random.bit_generator.state}')
# pre-env.seed np_random state: {'bit_generator': 'PCG64', 'state': {'state': 274959722186620119121085379683205939769, 'inc': 222618813376585624389888442161465505275}, 'has_uint32': 0, 'uinteger': 0}
env.seed(1)
print(f'pre-env.seed np_random state: {env._np_random.bit_generator.state}')
# pre-env.seed np_random state: {'bit_generator': 'PCG64', 'state': {'state': 274959722186620119121085379683205939769, 'inc': 222618813376585624389888442161465505275}, 'has_uint32': 0, 'uinteger': 0}
```

**Comment by RedTachyon** [comment]
For the record, in case someone stumbles upon this - it was more or less a false alarm, `env.seed` does what it's supposed to do, it's just that a few parts of attribute forwarding through wrappers were unintuitive, so it kinda looked like it didn't work if you squint in just the right way.

I think this can be closed? To avoid confusion, and the underlying issue has been fixed

**Comment by pseudo-rnd-thoughts** [comment]
@RedTachyon is correct, this was not a bug in `seed` but rather an issue with gym wrappers that would not pass `_np_random` to the actual wrapper which has been fixed in v24.1.
